tests/libvmi/annotation_test.go

```diff
@@ -0,0 +1,39 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ *
+ */
+
+package libvmi
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("libvmi annotations", func() {
```

Comment:
While I think it is fine to use unit tests to develop the test helpers themselves, I am less in favor of continuously testing the test helpers (i.e. letting them in with the merge). Feels like a recursion to me.

On top of that, I would also suggest never to have unit tests in the same package as the tested code. Testing non-exported items causes tests to be fragile and break frequently (implementation changes of a unit are reflected back to the test, even if they are "private" and not a unit API).

Reply:
Can you elaborate on the first point? IIUC this is the case for framework/matchers also, or what am I missing here?

To the second point I strongly disagree. Especially when writing unit tests it's important to test at the lowest level. Besides that, this is also the case in `framework/matchers`.